src/components/tasks/task-dialog.tsx

```diff
@@ -121,6 +159,121 @@ const TaskDialog = ({ projectId, milestoneId, open, task, onClose }: Props) => {
+   */
+  const createTaskConnectionsMutation = useMutation({
+    mutationFn: (params: CreateTaskConnectionRequest) => taskConnectionsApi.createTaskConnection(params),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["taskConnections", projectId, task?.id] });
+    },
+    onError: (error) => console.error(t("errorHandling.errorCreatingTaskConnection"), error),
+  });
+
+  /**
+   * Update task connections mutation
+   */
+  const updateTaskConnectionsMutation = useMutation({
+    mutationFn: (params: UpdateTaskConnectionRequest) => taskConnectionsApi.updateTaskConnection(params),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["taskConnections", projectId, task?.id] });
+    },
+    onError: (error) => console.error(t("errorHandling.errorUpdatingTaskConnection"), error),
+  });
+
+  /**
+   * Delete task connections mutation
+   */
+  const deleteTaskConnectionsMutation = useMutation({
+    mutationFn: (params: DeleteTaskConnectionRequest) => taskConnectionsApi.deleteTaskConnection(params),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["taskConnections", projectId, task?.id] });
+    },
+    onError: (error) => console.error(t("errorHandling.errorDeletingTaskConnection"), error),
+  });
+
+  /**
+   * Persist new and edited task connections
+   *
+   * @param taskId task id
+   */
+  const persistNewAndEditedTaskConnections = async (taskId: string) => {
+    const newConnections = newTaskConnections.map((connection) => ({
+      projectId: projectId,
+      taskConnection: {
+        sourceTaskId:
+          connection.hierarchy === TaskConnectionRelationship.CHILD ? taskId : connection.attachedTask?.id ?? "",
+        targetTaskId:
+          connection.hierarchy === TaskConnectionRelationship.CHILD ? connection.attachedTask?.id ?? "" : taskId,
+        type: connection.type,
+      },
+    }));
+
+    const editedConnections = existingTaskConnections.map((connection) => ({
+      projectId: projectId,
+      connectionId: connection.connectionId ?? "",
+      taskConnection: {
+        sourceTaskId:
+          connection.hierarchy === TaskConnectionRelationship.CHILD ? taskId : connection.attachedTask?.id ?? "",
+        targetTaskId:
+          connection.hierarchy === TaskConnectionRelationship.CHILD ? connection.attachedTask?.id ?? "" : taskId,
+        type: connection.type,
+      },
+    }));
+
+    const connectionsToDelete = (listTaskConnectionsQuery.data ?? [])
+      .filter((connection) => !existingTaskConnections.some((c) => c.connectionId === connection.id))
+      .map((connection) => ({ projectId, connectionId: connection.id ?? "" }));
+
+    await Promise.all([
+      ...newConnections.map((params) => createTaskConnectionsMutation.mutateAsync(params)),
+      ...editedConnections.map((params) => updateTaskConnectionsMutation.mutateAsync(params)),
+      ...connectionsToDelete.map((params) => deleteTaskConnectionsMutation.mutateAsync(params)),
```

Comment:
This might be okay as a shorthand in this case?
```suggestion
      ...newConnections.map(createTaskConnectionsMutation.mutateAsync),
      ...editedConnections.map(updateTaskConnectionsMutation.mutateAsync),
      ...connectionsToDelete.map(deleteTaskConnectionsMutation.mutateAsync),
```